src/main/kotlin/com/zopa/ktor/opentracing/Span.kt

```diff
@@ -2,9 +2,21 @@ package com.zopa.ktor.opentracing
 
 import io.opentracing.Span
 
+// cannot be internal see https://youtrack.jetbrains.com/issue/KT-21178 for context
+lateinit var tagsToAdd: List<Pair<String, () -> String>>
+
 inline fun <T> span(name: String = "defaultSpanName", block: Span.() -> T): T {
     val tracer = getGlobalTracer()
     val span = tracer.buildSpan(name).start()
+
+    tagsToAdd.forEach {
+        try {
+            span.setTag(it.first, it.second.invoke())
+        } catch (e: Exception) {
+            log.warn(e) { "Could not add tag: ${it.first}" }
+        }
+    }
```

Comment:
can you maybe extract this into a function as such ; https://github.com/MicroUtils/kotlin-logging/blob/master/src/commonMain/kotlin/mu/internal/MessageInvoker.kt so that you don't have to repeat the try / catch

Reply:
how about an extension function on `Span` ?

Reply:
As discussed it would also have to be made public because of the constraint for inline functions, so keeping as is.